.github/workflows/docker.yml

```diff
@@ -22,80 +24,67 @@ concurrency:
+  test-amd64:
+    needs: build-amd64
+    uses: ./.github/workflows/docker-test.yml
+    strategy:
+      matrix:
+        target: [aiida-core-base, aiida-core-with-services, aiida-core-dev]
     with:
-      architecture: arm64
-      runsOn: ARM64
-    if: ${{ !github.event.pull_request.head.repo.fork }}
+      runsOn: ubuntu-22.04
+      images: ${{ needs.build-amd64.outputs.images }}
+      target: ${{ matrix.target }}
 
-  amd64-push-ghcr:
-    uses: ./.github/workflows/docker-push.yml
+  build:
+    needs: test-amd64
+    uses: ./.github/workflows/docker-build.yml
     with:
-      architecture: amd64
-      registry: ghcr.io
-    secrets:
-      REGISTRY_USERNAME: ${{ github.actor }}
-      REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    needs: [amd64-build]
-    if: ${{ !github.event.pull_request.head.repo.fork }}
-
-  arm64-push-ghcr:
-    uses: ./.github/workflows/docker-push.yml
-    with:
-      architecture: arm64
-      registry: ghcr.io
-    secrets:
-      REGISTRY_USERNAME: ${{ github.actor }}
-      REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    needs: [arm64-build]
-    if: ${{ !github.event.pull_request.head.repo.fork }}
+      runsOn: ubuntu-22.04
+      platforms: linux/amd64,linux/arm64
 
-  merge-tags-ghcr:
-    uses: ./.github/workflows/docker-merge-tags.yml
+  publish-ghcr:
+    needs: [build, test-amd64]
+    uses: ./.github/workflows/docker-publish.yml
+    secrets: inherit
     with:
+      runsOn: ubuntu-22.04
+      images: ${{ needs.build.outputs.images }}
       registry: ghcr.io
-    secrets:
-      REGISTRY_USERNAME: ${{ github.actor }}
-      REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    needs: [amd64-push-ghcr, arm64-push-ghcr]
-    if: ${{ !github.event.pull_request.head.repo.fork }}
 
-  amd64-push-dockerhub:
-    if: github.repository == 'aiidateam/aiida-core' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
-    uses: ./.github/workflows/docker-push.yml
+  # IMPORTANT: To save arm64 runners resources,
+  # we run the test only when pushing to main.
+  # We also only test the aiida-core-dev image.
+  test-arm64:
+    needs: build
+    if: >-
+      github.repository == 'aiidateam/aiida-core'
+      && (github.ref_type == 'tag' || github.ref_name == 'main')
+    uses: ./.github/workflows/docker-test.yml
     with:
-      architecture: amd64
-      registry: docker.io
-    secrets:
-      REGISTRY_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-      REGISTRY_TOKEN: ${{ secrets.DOCKER_TOKEN }}
-    needs: [amd64-build]
-
-  arm64-push-dockerhub:
-    if: github.repository == 'aiidateam/aiida-core' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
-    uses: ./.github/workflows/docker-push.yml
-    with:
-      architecture: arm64
-      registry: docker.io
-    secrets:
-      REGISTRY_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-      REGISTRY_TOKEN: ${{ secrets.DOCKER_TOKEN }}
-    needs: [arm64-build]
+      runsOn: buildjet-4vcpu-ubuntu-2204-arm
```

Comment:
Hi @danielhollas, I just talk to the guy from warpBuild, and it has $8 per month for free, may try once by just change to `warp-ubuntu-latest-arm64-2x`?

Reply:
Sure, I'll give it a go. I am just going out the office, so will do later today or tomorrow. (Or feel free to push to this branch if you can).